sentry/src/event_reducer.rs

```diff
@@ -0,0 +1,101 @@
+use primitives::sentry::{AggregateEvents, Event, EventAggregate};
+use primitives::Channel;
+
+// @TODO: Remove attribute once we use this function!
+#[allow(dead_code)]
+pub(crate) fn reduce(channel: &Channel, initial_aggr: &mut EventAggregate, ev: &Event) {
+    match ev {
+        Event::Impression { publisher, .. } => {
+            let impression = initial_aggr.events.get("IMPRESSION");
+
+            let merge = merge_impression_ev(impression, &publisher, &channel);
+
+            initial_aggr.events.insert("IMPRESSION".to_owned(), merge);
+        }
+        Event::Close => {
+            let creator = channel.creator.clone();
+            let close_event = AggregateEvents {
+                event_counts: vec![(creator.clone(), 1.into())].into_iter().collect(),
+                event_payouts: vec![(creator, channel.deposit_amount.clone())]
+                    .into_iter()
+                    .collect(),
+            };
+            initial_aggr.events.insert("CLOSE".to_owned(), close_event);
+        }
+        _ => {}
+    };
+}
+
+fn merge_impression_ev(
+    impression: Option<&AggregateEvents>,
+    earner: &str,
+    channel: &Channel,
+) -> AggregateEvents {
+    let mut impression = impression
+        .map(Clone::clone)
+        .unwrap_or_else(Default::default);
+
+    let event_counts = impression
+        .event_counts
+        .entry(earner.into())
+        .or_insert_with(|| 0.into());
+    *event_counts += &1.into();
+
+    let event_payouts = impression
+        .event_payouts
+        .entry(earner.into())
+        .or_insert_with(|| 0.into());
+    *event_payouts += &channel.spec.min_per_impression;
```

Comment:
There seems to be a condition missing it's 
```
new BN(channel.spec.minPerImpression || 1)
```

https://github.com/AdExNetwork/adex-validator/blob/05-stable/services/sentry/lib/eventReducer.js#L47

Reply:
It's not an `Option`, I am not sure what is the intend of `|| 1` exactly - if there is a value or if it's `0` then use `1`, can you give me details about that?

Reply:
if `minPerImpression` is undefined use 1 the default price

Reply:
We do not use an `Option` in the Spec for the `minPerImpression` so we don't need this check. It will always be `>= 0` as the `Bignum` allows only unsigned numbers